nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -205,51 +209,184 @@ def init_model_parallel(self, global_rank: int, world_size: int) -> None:
                 if app_state.init_mpi_proc_group:
                     torch.distributed.new_group(backend='mpi')
 
+    def optimizer_sharded_state_dict(self):
```

Comment:
this logic is `MainParamsOptimizerWrapper`, should it be implemented as part of `MainParamsOptimizerWrapper`? DistOpt will have a different logic